src/agent/io/sourcemapper.ts

```diff
@@ -246,13 +259,14 @@ export class SourceMapper {
 }
 
 export function create(
+    prefix: string|undefined,
```

Comment:
prefer `prefix` was never undefined.

I see that you needed to modify all the tests to pass in `undefined` by default. That isn't very readable. My preference would be to add an overload that accepts the prefix string and then uses type inspection, or argument count checks, to figure out whether a prefix was provided.

In the rest of this file, i.e. in the constructor, enforce that prefix is never undefined.